### PR TITLE
Add missing features to new feed importer

### PR DESCRIPTION
**Pull request by rhymes**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

This adds the remaining missing features to the new feed importer:

- `Feeds::ImportArticlesWorker` which will eventually replace `Articles::RssReaderWorker` and `RssReaderFetchUserWorker`
- import of a single user's feed url
- `Feeds::ValidateUrl` service to do just that
- a feature flag to switch from the `RSSReader` to the new version

## Related Tickets & Documents

#10998 

## Added tests?

- [x] Yes
- [ ] No, and this is why: _please replace this line with details on why tests
      have not been included_
- [ ] I need help with writing tests

## Added to documentation?

- [ ] [Developer Docs](https://docs.forem.com) and/or [Admin Guide](https://forem.gitbook.io/forem-admin-guide/)
- [ ] README
- [x] No documentation needed

## [optional] Are there any post deployment tasks we need to perform?

We could start activating this on smaller Forems that have feeds (do we know if there are any?) before activating it on DEV or just go ahead and turn it on for a while before we remove the old code

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -40,7 +40,10 @@ def update
     set_current_tab(params["user"]["tab"])
 
     if @user.update(permitted_attributes(@user))
-      RssReaderFetchUserWorker.perform_async(@user.id) if @user.feed_url.present?
+      # NOTE: [@rhymes] this queues a job to fetch the feed each time the profile is updated, regardless if the user
+      # explicitly requested "Feed fetch now" or simply updated any other field
```
Comment (rhymes) [inline app/controllers/users_controller.rb:44]: I added this comment mostly for explanatory reasons, I didn't know we were fetching the feed each time until I set out to add the feature flag below.

I think we can split it out to its own controller action but not in this PR

File: app/models/user.rb [hunk 2]
```diff
@@ -598,9 +598,16 @@ def conditionally_validate_summary
...
-    errors.add(:feed_url, "is not a valid RSS/Atom feed")
+    valid = if FeatureFlag.enabled?(:feeds_import)
+              Feeds::ValidateUrl.call(feed_url)
+            else
+              RssReader.new.valid_feed_url?(feed_url)
+            end
+
+    errors.add(:feed_url, "is not a valid RSS/Atom feed") unless valid
+  rescue StandardError => e
+    errors.add(:feed_url, e.message)
```
Comment (rhymes) [inline app/models/user.rb:610]: better safe than sorry :D

File: app/services/feeds/validate_url.rb [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    end
+
+    def initialize(feed_url)
+      @feed_url = feed_url
+    end
+
+    def call
+      return false if feed_url.blank?
+
+      xml = HTTParty.get(feed_url, timeout: 10).body
```
Comment (rhymes) [inline app/services/feeds/validate_url.rb:14]: 10 seconds is definitely generous, but better than no timeout

**Review by citizen428 — APPROVED** [review]
Like the feature flag use, 👍

**Comment by citizen428** [comment]
@rhymes Sorry, I noticed this approval may have been a bit premature, please feel free to re-request if you make substantial changes.

**Comment by rhymes** [comment]
@citizen428 nope, the only thing I forgot is the feature flag in the Rake task I think. I'm going to add that. Thanks for the review!

File: app/controllers/users_controller.rb [hunk 4]
```diff
@@ -325,4 +328,12 @@ def config_changed?
   def destroy_request_in_progress?
     Rails.cache.exist?("user-destroy-token-#{@user.id}")
   end
+
+  def import_articles_from_feed(user)
+    return if user.feed_url.blank?
+
+    worker = FeatureFlag.enabled?(:feeds_import) ? Feeds::ImportArticlesWorker : RssReaderFetchUserWorker
```
Comment (joshpuetz) [inline app/controllers/users_controller.rb:335]: 🎏 👏

**Review by joshpuetz — APPROVED** [review]
LGTM!

File: app/workers/articles/rss_reader_worker.rb [hunk 5]
```diff
@@ -5,13 +5,12 @@ class RssReaderWorker
     sidekiq_options queue: :medium_priority, retry: 10
 
     def perform
-      # Temporary
-      # @sre:mstruve This is temporary until we have an efficient way to handle this job
-      # for our large DEV community. Smaller Forems should be able to handle it no problem
-      return if SiteConfig.community_name == "DEV"
```
Comment (mstruve) [inline app/workers/articles/rss_reader_worker.rb:11]: 🙏  THANK YOU!

File: spec/requests/user/user_settings_spec.rb [hunk 6]
```diff
@@ -254,6 +254,32 @@ def send_request(export_requested: true)
         end
       end
     end
+
+    context "when requesting a fetch of the feed", vcr: { cassette_name: "feeds_import_medium_vaidehi" } do
+      let(:feed_url) { "https://medium.com/feed/@vaidehijoshi" }
```
Comment (vaidehijoshi) [inline spec/requests/user/user_settings_spec.rb:259]: OH LOOK WHO IT IS

**Review by vaidehijoshi — APPROVED** [review]
Really love the removal of hardcoded DEV stuff _plus_ the use of the feature flag in this PR.
🌟🌟🌟  gold stars from me!

File: spec/requests/user/user_settings_spec.rb (hunk 6 above)
Comment (rhymes) [reply spec/requests/user/user_settings_spec.rb:259]: hahahaha :D

**Review by benhalpern — APPROVED** [review]

**Comment by rhymes** [comment]
Set temporarily as draft to discuss the rollout plan

**Review by mstruve — APPROVED** [review]

**Pull request merged**